### ISSUE TITLE
TSLint flags error when initialize an object of Tagged Union Types in WebStorm

### ISSUE THREAD

**Issue by Guoli-Lyu** [palantir/tslint#3758]
### Bug Report
TSLint flags error when initialize an object of Tagged Union Types while Typescript compiler can compile my codes without any error or warning. 

**I find that in Visual Studio Code, TSLint behaves as expected.**

Is this a bug WebStorm or misconfiguration?

- __TSLint version__:  5.9.1
- __TypeScript version__: 2.7.2
- __Running TSLint via__: NodeJS API (Webstorm)

#### TypeScript code being linted

```ts
interface InterfaceA {
    type: 'A';
    text: string;
}

interface InterfaceB {
    type: 'B';
    index: number;
}
type UnionInterface = InterfaceA | InterfaceB;
const a: UnionInterface = {
    type: 'A',
    text: 'meow'
};
```

with `tslint.json` configuration:

```json
{
  "rulesDirectory": [
    "node_modules/codelyzer"
  ],
  "rules": {
    "array-type": [
      true,
      "array-simple"
    ],
    "ban": [
      true,
      {
        "name": "parseInt",
        "message": "tsstyle#type-coercion"
      },
      {
        "name": "parseFloat",
        "message": "tsstyle#type-coercion"
      },
      {
        "name": "Array",
        "message": "tsstyle#array-constructor"
      }
    ],
    "ban-types": [
      true,
      [
        "Object",
        "Use {} instead."
      ],
      [
        "String",
        "Use 'string' instead."
      ],
      [
        "Number",
        "Use 'number' instead."
      ],
      [
        "Boolean",
        "Use 'boolean' instead."
      ]
    ],
    "interface-name": [
      true,
      "never-prefix"
    ],
    "jsdoc-format": true,
    "new-parens": true,
    "no-angle-bracket-type-assertion": true,
    "no-any": true,
    "no-conditional-assignment": true,
    "no-default-export": true,
    "no-duplicate-variable": true,
    "no-namespace": [
      true,
      "allow-declarations"
    ],
    "no-reference": true,
    "object-literal-shorthand": true,
    "only-arrow-functions": [
      true,
      "allow-declarations",
      "allow-named-functions"
    ],
    "switch-default": true,
    "use-isnan": true,
    "arrow-return-shorthand": true,
    "callable-types": true,
    "class-name": true,
    "comment-format": [
      true,
      "check-space"
    ],
    "curly": true,
    "deprecation": {
      "severity": "warn"
    },
    "eofline": true,
    "forin": true,
    "import-blacklist": [
      true,
      "rxjs",
      "rxjs/Rx"
    ],
    "import-spacing": true,
    "indent": [
      true,
      "spaces"
    ],
    "interface-over-type-literal": true,
    "label-position": true,
    "max-line-length": [
      true,
      140
    ],
    "member-access": false,
    "member-ordering": [
      true,
      {
        "order": [
          "static-field",
          "instance-field",
          "static-method",
          "instance-method"
        ]
      }
    ],
    "no-arg": true,
    "no-bitwise": true,
    "no-console": [
      true,
      "debug",
      "info",
      "time",
      "timeEnd",
      "trace"
    ],
    "no-construct": true,
    "no-debugger": true,
    "no-duplicate-super": true,
    "no-empty": false,
    "no-empty-interface": true,
    "no-eval": true,
    "no-inferrable-types": [
      true,
      "ignore-params"
    ],
    "no-misused-new": true,
    "no-non-null-assertion": true,
    "no-shadowed-variable": true,
    "no-string-literal": false,
    "no-string-throw": true,
    "no-switch-case-fall-through": true,
    "no-trailing-whitespace": true,
    "no-unnecessary-initializer": true,
    "no-unused-expression": true,
    "no-use-before-declare": true,
    "no-var-keyword": true,
    "object-literal-sort-keys": false,
    "one-line": [
      true,
      "check-open-brace",
      "check-catch",
      "check-else",
      "check-whitespace"
    ],
    "prefer-const": true,
    "quotemark": [
      true,
      "single"
    ],
    "radix": true,
    "semicolon": [
      true,
      "always"
    ],
    "trailing-comma": [
      true,
      {
        "multiline": "always",
        "singleline": "never"
      }
    ],
    "triple-equals": [
      true,
      "allow-null-check"
    ],
    "typedef-whitespace": [
      true,
      {
        "call-signature": "nospace",
        "index-signature": "nospace",
        "parameter": "nospace",
        "property-declaration": "nospace",
        "variable-declaration": "nospace"
      }
    ],
    "typeof-compare": true,
    "unified-signatures": true,
    "variable-name": false,
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-separator",
      "check-type"
    ],
    "directive-selector": [
      true,
      "attribute",
      [
        "app"
      ],
      "camelCase"
    ],
    "component-selector": [
      true,
      "element",
      [
        "app"
      ],
      "kebab-case"
    ],
    "no-output-on-prefix": true,
    "use-input-property-decorator": true,
    "use-output-property-decorator": true,
    "use-host-property-decorator": true,
    "no-input-rename": true,
    "no-output-rename": true,
    "use-life-cycle-interface": true,
    "use-pipe-transform-interface": true,
    "component-class-suffix": true,
    "directive-class-suffix": true
  }
}

```

#### Actual behavior

The following code can pass TSLint without complaint.
``` TypeScript
const b: InterfaceA = {
    type: 'A',
    text: 'meow'
};
```
However, when initialing tagged union type, TSlint complains that 
> Initializer type {type:"A",text: string} is not assignable to variable type UnionInterface

``` TypeScript
const c: UnionInterface = {
    type: 'A',
    text: 'meow'
};
```
Typescript Compile can compile above code without any error or warning.

#### Expected behavior
This is not an error, and thus TSLint should not complain at all.

**Comment by Guoli-Lyu** [comment]
This is a bug of WebStorm, and fixed in Early Access Program.
